torch/distributed/_composable/fsdp/fully_shard.py

```diff
@@ -1,21 +1,30 @@
-from typing import Any
+from typing import Any, Union
 
+import torch
 import torch.nn as nn
 
 from torch.distributed._composable import contract
 from torch.distributed._composable_state import _insert_module_state
 
+from ._fsdp_init import _normalize_device
+from ._fsdp_state import FSDPState
 
-@contract()
+
+@contract(state_cls=FSDPState)
 def fully_shard(
     module: nn.Module,
+    *,
+    device: Union[torch.device, int, str] = "cuda",
```

Comment:
Likewise, we have a TypeAlias: https://github.com/pytorch/pytorch/blob/af831415a876079a2ff63d72ae99d03e4c11aebd/torch/_prims_common/__init__.py#L55